### ISSUE TITLE
Remove @aws-sdk/client-secrets-manager dependency from backend

### ISSUE THREAD

**Issue by dangtony98** [Infisical/infisical#678]
### Feature description
As part of the initiative to reduce dependencies on the backend, it'd be great to remove the `@aws-sdk/client-secrets-manager` dependency that's currently used as an SDK in the AWS secret manager (AWS SM) integration to sync secrets from Infisical to AWS SM.

This task involves replacing this dependency with the HTTPS Query API as mentioned [here](https://docs.aws.amazon.com/secretsmanager/latest/userguide/asm_access.html#asm-sdks_query-api).

The HTTPS Query API gives you [programmatic access](https://docs.aws.amazon.com/secretsmanager/latest/apireference/Welcome.html) to Secrets Manager and AWS. The HTTPS Query API allows you to issue HTTPS requests directly to the service and thus we can use it to create/read/update/delete secrets in AWS SM.

### Why would it be useful?
This would further enhance the security of Infisical by reducing dependencies.

### Additional context
This shouldn't be too difficult and I'd recommend it as a great first issue especially since the AWS SM integration is already made; we just need to refactor logic to use the HTTP Query API instead of the AWS SM SDK.

**Comment by abdullahranginwala** [comment]
I would love to work on this issue. Based on my initial look at the code, I have identified the following services/functions to be replaced with HTTPS Query API:
`CreateSecretCommand`, `GetSecretValueCommand`, `ResourceNotFoundException`, `SecretsManagerClient`, `UpdateSecretCommand` . 

However, I couldn't find a detailed documentation for making the create/read/update/delete calls.

This seems like a straightforward implementation but some more context about the APIs will be extremely helpful.

**Comment by dangtony98** [comment]
That would be great @abdullahranginwala - Just assigned it to you!

Definitely join the Slack; best to communicate there

**Comment by abdullahranginwala** [comment]
I am unable to join slack since it requires a work email ending with @infisical.com

**Comment by JunedKhan101** [comment]
@abdullahranginwala I don't think so that is the case, I have joined the slack community with my gmail account.

**Comment by dangtony98** [comment]
Hi @abdullahranginwala,

Can you try joining via this link: https://infisical.com/slack

Let me know if it works!

**Comment by abdullahranginwala** [comment]
Yup, worked! I was trying to access the other link

**Comment by abdullahranginwala** [comment]
Apologies for such a late reply. Is this issue still active? @dangtony98

**Comment by abdullahranginwala** [comment]
Upon further investigation, it seems the issue is more complex due to AWS Signature 4 authentication requirements. For now, I am setting this aside.

**Comment by dangtony98** [comment]
@abdullahranginwala Sounds good!

**Comment by sean3687** [comment]
Is anyone working on this issue? Can I work on it?

**Comment by Siddharth-2382** [comment]
Hey @dangtony98 can you get someone to review the PR and test it. Let me know if any changes are required. Thanks ;)

**Comment by prayanshchh** [comment]
is this issue still active?
@dangtony98

**Comment by akhilmhdh** [comment]
Closing as removed from our roadmap.
